[GStreamer] Fix a failure of fast/speechsynthesis/voices-non-mock.html

shivamidow opened the pull request:
#### 3aad9165b15df302e611f79f33378137101d390f
<pre>
[GStreamer] Fix a failure of fast/speechsynthesis/voices-non-mock.html
<a href="https://bugs.webkit.org/show_bug.cgi?id=250656">https://bugs.webkit.org/show_bug.cgi?id=250656</a>

Reviewed by Philippe Normand.

The voices-non-mock.html used a mac platform-specific value to test the number
of available voices. This change introduces a new method of Internals, which allows
each port to decide its minimum expected voice count. Besides, this change includes
a new expected result for ports using GStreamer since the number of available voices
for those ports differs from that of the mac port.

* LayoutTests/fast/speechsynthesis/voices-non-mock.html:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/fast/speechsynthesis/voices-non-mock-expected.txt: Added.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::minimumExpectedVoiceCount):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: <a href="https://commits.webkit.org/259820@main">https://commits.webkit.org/259820@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/91a4bced8a1532d1e7bb5100b3228f300995d6a5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105702 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14792 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38574 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114926 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175067 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109603 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16199 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5989 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97965 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114731 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111455 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12320 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95303 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39797 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94184 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26943 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/81514 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8058 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28296 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8553 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/4883 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14174 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47843 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6795 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10106 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on LayoutTests/fast/speechsynthesis/voices-non-mock.html:
```diff
@@ -23,7 +23,7 @@
     }
 
-    shouldBeTrue("voiceCount > 20");
+    shouldBeTrue(`voiceCount > ${window.internals.expectedVoiceCount}`);
```
How about:

```suggestion
    shouldBeTrue(`voiceCount >= ${window.internals.expectedVoiceCount}`);
```

Or even:

```suggestion
    shouldBeTrue(`voiceCount >= ${window.internals.minimumExpectedVoiceCount}`);
```

shivamidow replied:
Why not? =) Let me come back with an updated patch.

shivamidow committed:
[GStreamer] Fix a failure of fast/speechsynthesis/voices-non-mock.html
https://bugs.webkit.org/show_bug.cgi?id=250656

Reviewed by Philippe Normand.

The voices-non-mock.html used a mac platform-specific value to test the number
of available voices. This change introduces a new method of Internals, which allows
each port to decide its minimum expected voice count. Besides, this change includes
a new expected result for ports using GStreamer since the number of available voices
for those ports differs from that of the mac port.

* LayoutTests/fast/speechsynthesis/voices-non-mock.html:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/fast/speechsynthesis/voices-non-mock-expected.txt: Added.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::minimumExpectedVoiceCount):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: https://commits.webkit.org/259820@main
